Index: suds/mx/literal.py
===================================================================
--- suds/mx/literal.py	(revision 588)
+++ suds/mx/literal.py	(working copy)
@@ -148,7 +148,7 @@
         name = resolved.name
         if self.options.xstq:
             ns = resolved.namespace('ns1')
-        Typer.manual(node, name, ns)
+        #Typer.manual(node, name, ns)
     
     def skip(self, content):
         """ skip this content """
