diff --git a/dslib/suds-svn-work/suds/transport/http.py b/dslib/suds-svn-work/suds/transport/http.py
index 8f1c506..4f99dc4 100644
--- a/dslib/suds-svn-work/suds/transport/http.py
+++ b/dslib/suds-svn-work/suds/transport/http.py
@@ -83,6 +83,8 @@ class HttpTransport(Transport):
         self.cookiejar = CookieJar()
         self.urlopener = u2.build_opener(SUDSHTTPRedirectHandler(),
                                          u2.HTTPCookieProcessor(self.cookiejar))
+        if self.options.proxy:
+          self.urlopener.add_handler(u2.ProxyHandler(self.options.proxy))        
 
     def open(self, request):
         try:
@@ -111,6 +113,9 @@ class HttpTransport(Transport):
             self.__setproxy(url, u2request)
             request.headers.update(u2request.headers)
             log.debug('sending:\n%s', request)
+            if self.options.proxy:
+              if not u2request._tunnel_host and u2request.origin_req_host != u2request.host:
+                u2request._tunnel_host = u2request.origin_req_host
             fp = self.__open(u2request)
             self.__getcookies(fp, u2request)
             result = Reply(200, fp.headers.dict, fp.read())
diff --git a/dslib/suds-svn-work/suds/transport/https.py b/dslib/suds-svn-work/suds/transport/https.py
index 3c3a4a2..2427640 100644
--- a/dslib/suds-svn-work/suds/transport/https.py
+++ b/dslib/suds-svn-work/suds/transport/https.py
@@ -58,7 +58,7 @@ class HttpAuthenticated(HttpTransport):
         HttpTransport.__init__(self, **kwargs)
         self.pm = u2.HTTPPasswordMgrWithDefaultRealm()
         self.handler = u2.HTTPBasicAuthHandler(self.pm)
-        self.urlopener = u2.build_opener(self.handler)
+        self.urlopener.add_handler(self.handler)
         
     def open(self, request):
         credentials = self.credentials()
